File: fortran_codes/gaussqdr.f90
program gaussqdr
  implicit none

  real :: a, b, x, C, t, I = 0
  integer :: n, ii

  open(unit = 1, file = "../input/gaussqdr.in", status = 'old')
  read(1,*) n
  read(1,*) a, b

  do ii = 1,n-1
    read(1,*) C, t
    x = (b-a)*t/2 + (b+a)/2
    I = I + C*func(x)*(b-a)/2
  end do
  
  print *, "The integration of the above function under limits", a, "and", b, "is:", I

contains

real function func (x)
  implicit none
  real :: x
  func = 2*x**2 -3*x
end function func

end program gaussqdr

File: cpp_codes/gaussqdr.cc
#include <iostream>
#include <fstream>
#include <cmath>

double func (double x) {
  double func = x;
  return func;
}

int main () {

  int n;
  double I = 0, a, b;

  cin >> a >> b;

  ifstream infile("gaussqdr.in");
  infile >> n;

  double* C = new double[n];
  double* t = new double[n];

  for (int i = 0; i < n; i++) {

    infile >> C[i] >> t[i];
    double x = (b-a)*t[i]/2 + (b+a)/2;
    I = I + C[i]*func(x);
  }

  cout << "The integration of the above function under limits " << a <<  " and " << b << " is: " << I << endl;
}